snapcraft/internal/project_loader/grammar/_processor.py

```diff
@@ -98,9 +97,14 @@ def process(
                 if _ELSE_FAIL_PATTERN.match(section):
                     _handle_else(statement, None)
                 else:
-                    primitives.add(self._transformer(call_stack, section, self.project))
+                    primitive = self._transformer(call_stack, section, self.project)
+                    primitives.append(primitive)
             elif isinstance(section, dict):
-                statement = self._parse_dict(section, statement, statements, call_stack)
+                statement, primitive = self._parse_dict(
+                    section, statement, statements, call_stack
+                )
+                if primitive:
+                    primitives.append(primitive)
```

Comment:
I'm confused by this. The `primitive` here is not actually a primitive, [but a dict](https://github.com/snapcore/snapcraft/pull/3360/files#diff-e361c5daac4f8c36aef648b1fbbab2ee062f78f12de81898b70f26547db91f92R201) (or `None`, of course). Am I missing something?

Reply:
Mmm, I interpreted "primitive" usage here as the build-grammar primitive (which now includes dict), not so much a python primitive.  I can rewrite it if you have a better term :D